File: toolbardemo/src/main/java/com/example/toolbardemo/MainActivity.java
package com.example.toolbardemo;

import android.graphics.Bitmap;
import android.graphics.BitmapFactory;
import android.graphics.drawable.ColorDrawable;
import android.os.Bundle;
import android.view.Menu;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;
import androidx.palette.graphics.Palette;

import java.util.List;
import java.util.Objects;

public class MainActivity extends AppCompatActivity {

    private Toolbar mToolbar;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        mToolbar = findViewById(R.id.toolbar);
        mToolbar.setTitle("主标题");
        mToolbar.setSubtitle("子标题");
        mToolbar.setLogo(android.R.mipmap.sym_def_app_icon);
        setSupportActionBar(mToolbar);
        mToolbar.setNavigationIcon(android.R.drawable.ic_menu_view);

        mToolbar.setOnMenuItemClickListener(item -> {
            switch (item.getItemId()) {
                case R.id.action_share:
                    Toast.makeText(MainActivity.this, "Share", Toast.LENGTH_SHORT).show();
                    break;
                case R.id.action_settings:
                    Toast.makeText(MainActivity.this, "Settings", Toast.LENGTH_SHORT).show();
                    break;
                default:
                    break;
            }
            return true;
        });

        Bitmap bitmap = BitmapFactory.decodeResource(getResources(), R.mipmap.test);
        Palette.from(bitmap).generate(palette -> {
            assert palette != null;
            List<Palette.Swatch> swatches = palette.getSwatches();
            if (swatches.size() > 0) {
                Objects.requireNonNull(getSupportActionBar()).setBackgroundDrawable(new ColorDrawable(swatches.get(0).getRgb()));
            }
        });
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        getMenuInflater().inflate(R.menu.menu_main, menu);
        return true;
    }
}

File: kotlin/kotlindemo2/src/main/java/com/example/kotlindemo2/MainActivity.kt
package com.example.kotlindemo2

import android.media.MediaPlayer
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import kotlin.coroutines.*

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        var str:String = resources.getString(R.string.name,"666")

        val mediaPlayer = MediaPlayer()

        val continuation = suspend {
            5
        }.createCoroutine(object:Continuation<Int> {
            override val context: CoroutineContext
                get() = EmptyCoroutineContext

            override fun resumeWith(result: Result<Int>) {

            }
        })

        continuation.resume(Unit)

    }
}